src/java.base/share/classes/java/io/Console.java

```diff
@@ -145,6 +146,124 @@ public Reader reader() {
         throw newUnsupportedOperationException();
     }
 
+    /**
+     * Writes a string representation of the specified object to this console's
+     * output stream, terminates the line and then flushes the console.
+     *
+     * <p> The effects of {@code con.println(obj)} are as if by calling
+     * {@snippet lang = java:
+     *     con.format("%s%n", obj)
+     * }
+     *
+     * @param  obj
+     *         An object whose string representation is to write.
+     *
+     * @return  This console
+     *
+     * @since 23
+     */
+    @PreviewFeature(feature = PreviewFeature.Feature.IMPLICIT_CLASSES)
+    public Console println(Object obj) {
+        throw newUnsupportedOperationException();
+    }
+
+    /**
+     * Writes a string representation of the specified object to this console's
+     * output stream and then flushes the console.
+     *
+     * <p> The effects of {@code con.print(obj)} are as if by calling
+     * {@snippet lang = java:
+     *     con.format("%s", obj)
+     * }
+     *
+     * @param  obj
+     *         An object whose string representation is to write.
+     *
+     * @return  This console
+     *
+     * @since 23
+     */
+    @PreviewFeature(feature = PreviewFeature.Feature.IMPLICIT_CLASSES)
+    public Console print(Object obj) {
+        throw newUnsupportedOperationException();
+    }
+
+    /**
+     * Provides a prompt, then reads a single line of text from the console.
```

Comment:
Add statement about flushing behavior here.

Reply:
While we can be painfully explicit here, I think that this method's behaviour is obvious: the prompt is output without any additional action from the client. How can it be otherwise? `readln` is a blocking call: even if one needed to call `flush`, one wouldn't be able to do it until after `readln` has returned, but by that time any data would have been already read and no prompt would be needed.

Reply:
One approach would be to defer to other methods, something like
> Writes a prompt as if by calling `print`, then reads a single line of text from the console.
Then we can quibble about how to specify flushing behavior on the output methods. :-)

Reply:
> One approach would be to defer to other methods, something like
> 
> > Writes a prompt as if by calling `print`, then reads a single line of text from the console.
> 
> Then we can quibble about how to specify flushing behavior on the output methods. :-)

Done.